src/Controller/Api/Legacy/GroupController.php

```diff
@@ -0,0 +1,256 @@
+     *
+     * @View(serializerGroups={"sonata_api_read"}, serializerEnableMaxDepthChecks=true)
+     *
+     * @return PagerInterface
+     */
+    public function getGroupsAction(ParamFetcherInterface $paramFetcher)
+    {
+        $supportedFilters = [
+            'enabled' => '',
+        ];
+
+        $page = $paramFetcher->get('page');
+        $limit = $paramFetcher->get('count');
+        $sort = $paramFetcher->get('orderBy');
+        $criteria = array_intersect_key($paramFetcher->all(), $supportedFilters);
+
+        $criteria = array_filter($criteria, static function ($value): bool {
+            return null !== $value;
+        });
+
+        if (!$sort) {
+            $sort = [];
+        } elseif (!\is_array($sort)) {
+            $sort = [$sort, 'asc'];
+        }
+
+        return $this->groupManager->getPager($criteria, $page, $limit, $sort);
+    }
+
+    /**
+     * Retrieves a specific group.
+     *
+     * @ApiDoc(
+     *  requirements={
+     *      {"name"="id", "dataType"="string", "description"="Group identifier"}
+     *  },
+     *  output={"class"="FOS\UserBundle\Model\GroupInterface", "groups"={"sonata_api_read"}},
+     *  statusCodes={
+     *      200="Returned when successful",
+     *      404="Returned when group is not found"
+     *  }
+     * )
+     *
+     * @View(serializerGroups={"sonata_api_read"}, serializerEnableMaxDepthChecks=true)
+     *
+     * @param string $id
+     *
+     * @return GroupInterface
+     */
+    public function getGroupAction($id)
+    {
+        return $this->getGroup($id);
+    }
+
+    /**
+     * Adds a group.
+     *
+     * @ApiDoc(
+     *  input={"class"="sonata_user_api_form_group", "name"="", "groups"={"sonata_api_write"}},
+     *  output={"class"="Sonata\UserBundle\Model\Group", "groups"={"sonata_api_read"}},
+     *  statusCodes={
+     *      200="Returned when successful",
+     *      400="Returned when an error has occurred during the group creation",
+     *  }
+     * )
+     *
+     * @param Request $request A Symfony request
+     *
+     * @throws NotFoundHttpException
+     *
+     * @return GroupInterface
+     */
+    public function postGroupAction(Request $request)
+    {
+        return $this->handleWriteGroup($request);
+    }
+
+    /**
+     * Updates a group.
+     *
+     * @ApiDoc(
+     *  requirements={
+     *      {"name"="id", "dataType"="string", "description"="Group identifier"}
+     *  },
+     *  input={"class"="sonata_user_api_form_group", "name"="", "groups"={"sonata_api_write"}},
+     *  output={"class"="Sonata\UserBundle\Model\Group", "groups"={"sonata_api_read"}},
+     *  statusCodes={
+     *      200="Returned when successful",
+     *      400="Returned when an error has occurred during the group creation",
+     *      404="Returned when unable to find group"
+     *  }
+     * )
+     *
+     * @param string  $id      Group identifier
+     * @param Request $request A Symfony request
```

Comment:
Not sure, the comment is really useful

Reply:
https://github.com/sonata-project/SonataUserBundle/pull/1225#discussion_r493025701